src/network/networkd-link.c

```diff
@@ -1000,8 +998,6 @@ static int route_handler(sd_netlink *rtnl, sd_netlink_message *m, Link *link) {
 
         assert(link);
         assert(link->route_messages > 0);
-        assert(IN_SET(link->state, LINK_STATE_CONFIGURING,
-                      LINK_STATE_FAILED, LINK_STATE_LINGER));
```

Comment:
Why remove this ? Is the race still there

Reply:
`link_request_set_routes()` is called when dynamic addresses are configured. In that time, the link state may be already in configured state.

Reply:
This means the race is still there. We count the ```route_messages```  and when this becomes zero we make it configure.

Reply:
Sorry, I noticed that `link_request_set_routes()` calls `link_set_state(link, LINK_STATE_CONFIGURING)`. So it is not necessary to remove this assertion. Now this change is dropped. Thanks.